### PR TITLE
Added nfs and smartquota info

### PR DESCRIPTION
**Pull request by Kritika-Bhateja-03**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [ ] I have performed Ansible Sanity test using --docker default
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: plugins/modules/info.py [hunk 1]
```diff
@@ -3149,7 +3272,8 @@ def perform_module_operation(self):
             SnmpSettings=snmp_settings,
             ServerCertificate=server_certificate,
             roles=roles,
-            support_assist_settings=support_assist_settings
+            support_assist_settings=support_assist_settings,
+            Smartquota=smartquota
```
Comment (sachin-apa) [inline plugins/modules/info.py:3276]:
```suggestion
            smart_quota=smartquota
```

Comment (Kritika-Bhateja-03) [reply plugins/modules/info.py]: Done

File: plugins/modules/nfs.py [hunk 2]
```diff
@@ -196,9 +197,6 @@
             choices: [absent, present]
             default: present
 
-notes:
```
Comment (Bhavneet-Sharma) [inline plugins/modules/nfs.py:199]: Instead of removing this, better like this
**The I(check_mode) is supported.**

File: plugins/modules/nfs.py [hunk 3]
```diff
@@ -588,19 +586,21 @@ def _get_nfs_export_from_id(self, nfs_export_id, access_zone):
...
-                read_write_clients=self.module.params['read_write_clients'],
-                root_clients=self.module.params['root_clients'],
-                read_only=self.module.params['read_only'],
-                all_dirs=self.module.params['sub_directories_mountable'],
-                description=self.module.params['description'],
-                security_flavors=get_security_keys(
-                    self.module.params['security_flavors']),
-                zone=self.module.params['access_zone'])
-            return nfs_export
+            if not self.module.check_mode:
```
Comment (Bhavneet-Sharma) [inline plugins/modules/nfs.py:589]: Do we need check_mode condition here? Here this method does prepare the payload.

**Review by Bhavneet-Sharma — COMMENTED** [review]
Validate module files with Andersson's script as well.

File: plugins/modules/nfs.py (hunk 2 above)
Comment (Kritika-Bhateja-03) [reply plugins/modules/nfs.py:199]: done

File: plugins/modules/nfs.py (hunk 3 above)
Comment (Kritika-Bhateja-03) [reply plugins/modules/nfs.py]: handled this by removing.

**Comment by sachin-apa** [comment]
run e2e

**Comment by Kritika-Bhateja-03** [comment]
image
UT coverage for smartquota

**Review by sachin-apa — COMMENTED** [review]
Verified with automation and changes are good to go.
Please check UT the coverage and fix the UT failures.

**Review by sachin-apa — APPROVED** [review]

**Review by meenakshidembi691 — DISMISSED** [review]

**Review by Bhavneet-Sharma — DISMISSED** [review]

**Review by meenakshidembi691 — DISMISSED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by Bhavneet-Sharma — DISMISSED** [review]

File: tests/unit/plugins/modules/test_info.py [hunk 4]
```diff
@@ -640,6 +644,3 @@ def test_get_facts_event_groups_api_exception(self, gatherfacts_module_mock, gat
             gatherfacts_module_mock.perform_module_operation()
         assert MockGatherfactsApi.get_gather_facts_error_response(
             gather_subset) in gatherfacts_module_mock.module.fail_json.call_args[1]['msg']
-
```
Comment (Bhavneet-Sharma) [inline tests/unit/plugins/modules/test_info.py:643]: Why this has been removed?

**Review by sachin-apa — APPROVED** [review]

File: tests/unit/plugins/modules/test_info.py (hunk 4 above)
Comment (Kritika-Bhateja-03) [reply tests/unit/plugins/modules/test_info.py:643]: Just for testing few things, i had removed that piece of code. Added it back.

File: tests/unit/plugins/modules/test_info.py [hunk 5]
```diff
@@ -641,5 +645,84 @@ def test_get_facts_event_groups_api_exception(self, gatherfacts_module_mock, gat
...
+            'zone': "System",
+        })
+        gatherfacts_module_mock.module.params = self.get_module_args
+        with patch.object(gatherfacts_module_mock.namespace_api, MockGatherfactsApi.get_gather_facts_error_method(gather_subset)) as mock_method:
+            mock_method.side_effect = MagicMock(side_effect=MockApiException)
+            gatherfacts_module_mock.perform_module_operation()
+        assert MockGatherfactsApi.get_gather_facts_error_response(
+            gather_subset) in gatherfacts_module_mock.module.fail_json.call_args[1]['msg']
+
     def test_main(self):
```
Comment (sachin-apa) [inline tests/unit/plugins/modules/test_info.py:727]: what are we asserting here ?

**Review by sachin-apa — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

File: tests/unit/plugins/modules/test_filesystem.py [hunk 6]
```diff
@@ -1,199 +0,0 @@
-# Copyright: (c) 2022-2024, Dell Technologies
```
Comment (Bhavneet-Sharma) [inline tests/unit/plugins/modules/test_filesystem.py:1]: why this is deleted

**Review by trisha-dell — DISMISSED** [review]

**Review by Bhavneet-Sharma — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Review by trisha-dell — APPROVED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

**Pull request merged**